src/extern/driver.f90

```diff
@@ -117,11 +119,22 @@ subroutine singlepoint(self, env, mol, chk, printlevel, restart, &
     !$omp critical (turbo_lock)
     inquire (file='gradient', exist=exist)
     if (exist) then
+      ! ### only TM output is supported for now ###
       call rdtm(env,mol%n, .true., energy, gradient, xyz_cached)
       cache = all(abs(xyz_cached - mol%xyz) < 1.e-10_wp)
     end if
     if (.not. cache) then
-      call wrtm(mol%n, mol%at, mol%xyz)
+      call generateFileName(tmpname, 'xtbopt', extension, mol%ftype)
+      write(env%unit,'(/,a,1x,a,/)') &
+         "updated geometry written to:",tmpname
+      call open_file(ich,tmpname,'w')
+      if (exist) then
```

Comment:
This will only write the gradient if the geometry mismatches the one for the gradient calculation